From b7b732b69f66917ed08b800d5ec34394d6d2dcda Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Thu, 03 Oct 2024 12:16:17 +0300
Subject: [PATCH] [FSW-10021] fs1: configure spi dma restart crc

There are new bit fields in spi dma controls
to support new SPI DMA requirements. For the
timebeing to work with legacy style configure
them properly.

Change-Id: I37e7dfb512854bda48951ad2f7fa1b7994531ee4
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index cd67c39..74ed06c 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -328,7 +328,7 @@
 static int dma_copy_image_internal(uint16_t image_size, uint32_t flash_offset,
 				    uint64_t dest_offset, uint8_t fw_sram_dst)
 {
-	uint32_t spi_dma_ctrl =  0;
+	uint32_t spi_dma_ctrl =  0, restart_crc = 1;
 	uint32_t spi_dma_status;
 	uint32_t spi_dma_dst_addr_l = DMA_ADDRESS_LOW(dest_offset);
 	uint32_t spi_dma_dst_addr_h = DMA_ADDRESS_HIGH(dest_offset);
@@ -346,6 +346,10 @@
 	/* sram=1 hbw=0 as destination */
 	spi_dma_ctrl |= fw_sram_dst << PARC_GLOBAL_CONF_SPI_DMA_CTRL_DST_SRAM_S;
 
+	/* TODO: This patch is temporary until FSW-9792 is implemented */
+	/* 0: continue crc calc 1: restart crc */
+	spi_dma_ctrl |= restart_crc << PARC_GLOBAL_CONF_SPI_DMA_CTRL_CRC_RESTART_S;
+
 	/* # of 4bytes to DMA copy */
 	spi_dma_ctrl |= (uint32_t)image_size_4b << PARC_GLOBAL_CONF_SPI_DMA_CTRL_MEM_SIZE_S;
 
